### DISCUSSION TITLE
Include units in node configuration options

### DISCUSSION THREAD

**Discussion by NYZack** [zwave-js/zwave-js-ui#668] [Ideas]
Device configuration files include (optional) units for each configuration parameter, but these are not referenced in the Control Panel under the node Configuration options.  Therefore, in order to adequately describe a configuration parameter, the description field must include a mention of units.  This is redundant.  An easy fix might be to include units on the lines on which node configuration values are entered in the Control Panel:

[25-112-0-5] Motion Re-trigger Time
30  **_seconds_**
Delay before the sensor reports no motion after detecting the last motion activity. 0=set manually. (Default: 0)

**Comment by robertsLando** [comment]
Units should be displayed on the right side of each input

**Reply by NYZack** [reply]
Not sure if this is in the new release 1.2.3.  I see units for values that the device returns, but not for configuration options.

**Reply by robertsLando** [reply]
It means that configurations are missing the units in the device configuration file. This should be fixed on node-zwave-js, you could oopen an issue there and specify the affected devices

**Reply by NYZack** [reply]
I don't understand. My configuration files *do* have units, and they don't show up under configuration options on the Control Panel.

**Reply by robertsLando** [reply]
> My configuration files do have units

Could you send me a node export of the device that isn't showing units?

**Reply by NYZack** [reply]
Here are the config file and pasted "debug info" for a representative device.  Note that configuration parameter number 1 has "seconds" listed as units, but this doesn't show up in the debug info.

[zcombo-g.json.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6029617/zcombo-g.json.txt)
[debug info.txt](https://github.com/zwave-js/zwavejs2mqtt/files/6029620/debug.info.txt)

**Reply by NYZack** [reply]
And here's a screenshot showing that units are not shown.
![image](https://user-images.githubusercontent.com/8227920/108861277-08b0b800-75bd-11eb-890b-c62712970fae.png)

**Reply by robertsLando** [reply]
App version?

**Reply by robertsLando** [reply]
BTW:  


```
 {
      "id": "30-112-0-1",
      "nodeId": 30,
      "commandClass": 112,
      "commandClassName": "Configuration",
      "property": 1,
      "propertyName": "SuperVision report timeout",
      "type": "number",
      "readable": true,
      "writeable": true,
      "description": "ZCOMBO will send the message over SuperVision Command Class and it will wait for the SuperVision report from the Controller for the SuperVision report timeout time.",
      "label": "SuperVision report timeout",
      "default": 1500,
      "stateless": false,
      "min": 500,
      "max": 5000,
      "list": false,
      "value": 1500,
      "lastUpdate": 1614016233106,
      "newValue": 1500
    },
    {
      "id": "30-112-0-2",
      "nodeId": 30,
      "commandClass": 112,
      "commandClassName": "Configuration",
      "property": 2,
      "propertyName": "SuperVision retry count",
      "type": "number",
      "readable": true,
      "writeable": true,
      "description": "If the SuperVision report is not received within the SuperVision report timeout time, the ZCOMBO will retry sending the message again. Upon exceeding the max retry, the ZCOMBO device will send the next message available in the queue.",
      "label": "SuperVision retry count",
      "default": 1,
      "stateless": false,
      "min": 0,
      "max": 5,
      "list": false,
      "value": 1,
      "lastUpdate": 1614016233167,
      "newValue": 1
    },
    {
      "id": "30-112-0-3",
      "nodeId": 30,
      "commandClass": 112,
      "commandClassName": "Configuration",
      "property": 3,
      "propertyName": "SuperVision wait time",
      "type": "number",
      "readable": true,
      "writeable": true,
      "description": "Before retrying the message, ZCOMBO will wait for the SuperVision wait time. Actual wait time is calculated using the formula: Wait Time = SuperVision wait time base-value + random-value + (attempt-count x 5 seconds). The random value will be between 100 and 1100 milliseconds.",
      "label": "SuperVision wait time",
      "default": 5,
      "stateless": false,
      "min": 1,
      "max": 60,
      "list": false,
      "value": 5,
      "lastUpdate": 1614016233228,
      "newValue": 5
    }
```

Units are not present in values. @AlCalzone Is this an issue on zwavejs?

**Reply by NYZack** [reply]
> App version?

1.2.3

**Reply by AlCalzone** [reply]
Shit. I've bugged every contributor to add the units, but they never make it into the metadata

**Reply by AlCalzone** [reply]
Will fix later, should be included in tonights release.

**Reply by robertsLando** [reply]
Ok perfect it's not on my side :laughing: :heart:

**Reply by robertsLando** [reply]
Opened this: https://github.com/zwave-js/node-zwave-js/issues/1867

**Reply by NYZack** [reply]
Units don't seem to be displayed when there are options available for the configuration parameter.  For instance, the units here are "lux" for custom set values.  If there are options available, the user has no way of knowing what the units are.

![image](https://user-images.githubusercontent.com/8227920/111875492-72c83d00-8970-11eb-8733-048d2d2e97bf.png)

**Reply by robertsLando** [reply]
@NYZack did you try to re-interview your device?

**Reply by NYZack** [reply]
@robertsLando Yes, just tried again and waited for complete interview.  That doesn't work.

**Reply by robertsLando** [reply]
Fixed in https://github.com/zwave-js/zwavejs2mqtt/pull/925
